R/plotCCA.R

```diff
@@ -0,0 +1,504 @@
+setGeneric("plotCCA", signature = c("object"),
+           function(object, dimred, ...) standardGeneric("plotCCA"))
+
+#' @rdname plotRDA
+#' @aliases plotRDA
+#' @export
+setMethod("plotCCA", signature = c(object = "SingleCellExperiment"),
+    function(object, dimred, ...){
+        ###################### Input check #######################
+            
+        ###################### Input check end ####################
+        # Get data for plotting
+        plot_data <- .incorporate_rda_vis(object, dimred, ...)
+        # Create a plot
+        plot <- .rda_plotter(plot_data, ...)
+        return(plot)
+    }
+)
+
+#' @rdname plotRDA
+#' @aliases plotRDA
+#' @export
+setMethod("plotCCA", signature = c(object = "matrix"),
+    function(object, ...){
+        ###################### Input check #######################
+            
+        ###################### Input check end ####################
+        # Construct TreeSE from rda/cca object
+        object <- .rda2tse(object)
+        # Get data for plotting
+        plot_data <- .incorporate_rda_vis(object, "RDA", ...)
+        # Create a plot
+        plot <- .rda_plotter(plot_data, ...)
+        return(plot)
+    }
+)
+
+#' @rdname plotRDA
+#' @aliases plotCCA
+#' @export
+setGeneric("plotRDA", signature = c("object"),
+    function(object, dimred, ...) standardGeneric("plotRDA"))
+
+#' @rdname plotRDA
+#' @aliases plotCCA
+#' @export
+setMethod("plotRDA", signature = c(object = "SingleCellExperiment"),
+    function(object, dimred, ...){
+        ###################### Input check #######################
+        
+        ###################### Input check end ####################
+        # Get data for plotting
+        plot_data <- .incorporate_rda_vis(object, dimred, ...)
+        # Create a plot
+        plot <- .rda_plotter(plot_data, ...)
+        return(plot)
+    }
+)
+
+#' @rdname plotRDA
+#' @aliases plotCCA
+#' @export
+setMethod("plotRDA", signature = c(object = "matrix"),
+    function(object, ...){
+        ###################### Input check #######################
+            
+        ###################### Input check end ####################
+        # Construct TreeSE from rda/cca object
+        object <- .rda2tse(object)
+        # Get data for plotting
+        plot_data <- .incorporate_rda_vis(object, "RDA", ...)
+        # Create a plot
+        plot <- .rda_plotter(plot_data, ...)
+        return(plot)
+    }
+)
+
+
+################## HELP FUNCTIONS ##########################
+
+# Construct TreeSE from rda/cca object to pass it to downstream functions
+.rda2tse <- function(object) {
+    # If rda/cca object, convert it to TreeSE
+    if (is.matrix(object)) {
+        # Remove significance attribute if present
+        attr(rda, "significance") <- NULL
+        # Make empty colData
+        coldata <- data.frame(ID = rownames(rda))
+        rownames(coldata) <- coldata[["ID"]]
+        coldata[["ID"]] <- NULL
+        # Convert rda/cca object to TreeSE
```

Comment:
Since assay is only required one, you could make that to simplify the code

Something like this matrix(ncol = 3, dimnames = list(NULL, c("sam1", "sam2", "sam3")))